nixos/doc/manual/installation/installing-usb.section.md

```diff
@@ -1,31 +1,54 @@
-# Booting from a USB Drive {#sec-booting-from-usb}
+# Booting from a USB flash drive {#sec-booting-from-usb}
 
-For systems without CD drive, the NixOS live CD can be booted from a USB
-stick. You can use the `dd` utility to write the image:
-`dd if=path-to-image of=/dev/sdX`. Be careful about specifying the correct
-drive; you can use the `lsblk` command to get a list of block devices.
+The image has to be written verbatim to the USB flash drive for it to be
+bootable on UEFI and BIOS systems. Here are the recommended tools to do that.
 
-::: {.note}
-::: {.title}
-On macOS
+## Creating bootable USB flash drive with a graphical tool {#sec-booting-from-usb-graphical}
+
+Etcher is a popular and user-friendly tool. It works on Linux, Windows and macOS.
+
+Download it from [balena.io](https://www.balena.io/etcher/), start the program,
+select the downloaded NixOS ISO, then select the USB flash drive and flash it.
+
+::: {.warning}
+Etcher reports errors and usage statistics by default, which can be disabled in
+the settings.
 :::
 
+An alternative is [USBImager](https://gitlab.com/bztsrc/usbimager/#usbimager),
+which is very simple and does not connect to the internet. Download the version
+with write-only (wo) interface for your system. Start the program,
+select the image, select the USB flash drive and click "Write".
+
+## Creating bootable USB flash drive from a Terminal on Linux {#sec-booting-from-usb-linux}
+
+First, make sure all partitions on the device are properly unmounted. Then use
+the `dd` utility to write the image to the USB flash drive:
+
+```ShellSession
+sudo umount /dev/sdX*
+sudo dd if=<path-to-image> of=/dev/sdX bs=4M
```

Comment:
That also ensures that all data is flushed before ejecting the drive:

```suggestion
sudo dd if=<path-to-image> of=/dev/sdX bs=4M conv=fsync
```

Reply:
This flag does not exists on macos though. So only add it for linux.

Reply:
I generally use `oflag=sync` that seems to be similar, and I think it is available in macOS.

Reply:
No, `oflag=sync` is a valid alternative for `fsync` (yeah, the effect is different but it still works as "ensure that all data is flushed"): https://unix.stackexchange.com/a/508726

What I meant is that `oflag` is not available in macOS. So changing to it wouldn't solve anything.

Reply:
FWIW Busybox's `dd` supports `conv=fsync` but not `oflag`.

Would a standalone `sync` command suffice?

Reply:
I'm almost thinking at this point that it should be something like "add your preferred `dd` flags, if you're unsure, read the appropriate manual". As I stated, things are highly cargo-culted, and *always* causes this kind of discussion :). Though adding a final `sync` in that case would be beneficial to ensure that it's synced for people that won't look it up.

Reply:
`conv=fdatasync` and `conv=sync` ends-up being functionally identical for the requirement of syncing.

As would calling `sync` after `dd` (AFAIUI), or calling `eject`.